### PR TITLE
Story/Junits Rework

### PR DESCRIPTION
**Pull request by sgarg-CS**
This PR is in response to [PLUGIN-1371](https://cdap.atlassian.net/browse/PLUGIN-1371)

**Earlier:** 
image

**Now:**
image

### CODE REVIEW

**Review by rmstar — COMMENTED** [review]
Can you provide some context on what this PR is fixing? Is there a Jira associated with this change?

**Comment by sgarg-CS** [comment]
> Can you provide some context on what this PR is fixing? Is there a Jira associated with this change?

Yes, updated the same in the PR description.

File: src/test/java/io/cdap/plugin/successfactors/source/transport/RuntimeFunctionalTest.java [hunk 1]
```diff
@@ -174,9 +175,17 @@ public void verifyDataCorrectness()
         StructuredRecord structuredRecord = recordList.get(i);
         pluginSchema.getFields().forEach(field -> {
           String fieldName = field.getName();
-          Assert.assertEquals(field.getName() + " value is not equal.",
-                              processSchemaTypeValue(field.getSchema(), oDataEntry.getProperties().get(fieldName)),
-                              structuredRecord.get(field.getName()));
+          if (!field.getSchema().getType().isSimpleType() && field.getSchema().getUnionSchema(0).getLogicalType()
+            != null && field.getSchema().getUnionSchema(0).getLogicalType() == Schema.LogicalType.DECIMAL) {
+            Assert.assertArrayEquals(field.getName() + " value is not equal.",
+                                     (byte[]) processSchemaTypeValue(field.getSchema(), oDataEntry.getProperties()
```
Comment (rmstar) [inline src/test/java/io/cdap/plugin/successfactors/source/transport/RuntimeFunctionalTest.java:181]: Can you use `StructuredRecord.getDecimal()` here? https://github.com/cdapio/cdap/blob/594fc39bb863a68c05afa0da789ba9c3b9baacf2/cdap-api-common/src/main/java/io/cdap/cdap/api/data/format/StructuredRecord.java#L213

Comment (sgarg-CS) [reply src/test/java/io/cdap/plugin/successfactors/source/transport/RuntimeFunctionalTest.java:181]: If I use `StructuredRecord.getDecimal()`, then I'll have to modify the expected value as well from `pluginValue.unscaledValue().toByteArray()` to `pluginValue`


The method`StructuredRecord.Builder.setDecimal` itself returns the value in byte array form.
https://github.com/cdapio/cdap/blob/594fc39bb863a68c05afa0da789ba9c3b9baacf2/cdap-api-common/src/main/java/io/cdap/cdap/api/data/format/StructuredRecord.java#L507

**Review by rmstar — APPROVED** [review]

**Pull request merged**